core/state_processor.go

```diff
@@ -115,6 +117,24 @@ func applyTransaction(config *chain.Config, engine consensus.EngineReader, gp *G
 			ibs.GetState(types.L1BlockAddr, &types.L1BaseFeeSlot, &l1BaseFee)
 			ibs.GetState(types.L1BlockAddr, &types.OverheadSlot, &overhead)
 			ibs.GetState(types.L1BlockAddr, &types.ScalarSlot, &scalar)
+			//Bedrock
+			if l1CostFunc := evm.Context().L1CostFunc; l1CostFunc != nil {
+				l1Fee := l1CostFunc(evm.Context().BlockNumber, msg)
+				if l1Fee != nil {
+					receipt.L1Fee = l1Fee.ToBig()
+				} else {
+					receipt.L1Fee = &big.Int{}
+				}
+				feeScalar := new(big.Float).SetInt(scalar.ToBig())
+				receipt.FeeScalar = feeScalar.Quo(feeScalar, big.NewFloat(1e6))
+				l1GasUsed := uint256.NewInt(msg.RollupDataGas())
+				receipt.L1GasUsed = l1GasUsed.Add(l1GasUsed, &overhead).ToBig()
+				receipt.L1GasPrice = l1BaseFee.ToBig()
+				log.Debug("Set L1Fee for receipt", "fee", receipt.L1Fee, "feeScalar", feeScalar, "l1GasPrice", receipt.L1GasPrice, "l1GasUsed", receipt.L1GasUsed, "txhash", tx.Hash())
+			} else {
+				log.Warn("No L1 cost function set in context", "txhash", tx.Hash())
+			}
+			//Bedrock
```

Comment:
This logic can be removed

Reply:
let's remove this in a separate pr as I'm unsure of the consequences where this is used

Reply:
The logic of handling these special fields is added in this PR and the above code is removed. https://github.com/bobanetwork/v3-erigon/pull/30
I am not sure why this part of code is added back again.

Reply:
uff, you're right. my bad